swarm/src/lib.rs

```diff
@@ -1401,14 +1361,16 @@ where
 
     /// Builds a `Swarm` with the current configuration.
     pub fn build(mut self) -> Swarm<TBehaviour> {
-        let supported_protocols = self
-            .behaviour
-            .new_handler()
-            .inbound_protocol()
-            .protocol_info()
-            .into_iter()
-            .map(|info| info.protocol_name().to_vec())
-            .collect();
+        let supported_protocols = todo!();
+
+        // let supported_protocols = self
+        //     .behaviour
+        //     .new_handler()
+        //     .inbound_protocol()
+        //     .protocol_info()
+        //     .into_iter()
+        //     .map(|info| info.protocol_name().to_vec())
+        //     .collect();
```

Comment:
This turns out to be difficult.

@mxinden I remember you saying that `PollParameters` was always something that didn't appeal to you. Did you ever think about alternative design? This might be a good time to bring them up :)

Reply:
I am now initialising these to an empty vector and they are updated every time a new connection is made.

Reply:
> @mxinden I remember you saying that `PollParameters` was always something that didn't appeal to you.

Yes, very much. I would like to not have `PollParameters`.

> I am now initialising these to an empty vector and they are updated every time a new connection is made.

In my eyes that is the better behavior.

Reply:
> > @mxinden I remember you saying that `PollParameters` was always something that didn't appeal to you.
> 
> Yes, very much. I would like to not have `PollParameters`.

Tracking this here: https://github.com/libp2p/rust-libp2p/issues/3124